Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/ApplePushNotificationService/ApplePushNotificationServiceConnector.cs

```diff
@@ -39,11 +39,10 @@ public async Task<SendResults> Send(IEnumerable<PnsRegistration> registrations,
         {
             ValidateRegistration(pnsRegistration);
             var handle = pnsRegistration.Handle.Value;
-            var bundle = _options.GetBundleById(pnsRegistration.AppId!);
             var keyInformation = _options.GetKeyInformationForBundleId(pnsRegistration.AppId!);
             var jwt = _jwtGenerator.Generate(keyInformation.PrivateKey, keyInformation.KeyId, keyInformation.TeamId, pnsRegistration.AppId);
 
-            var request = new ApnsMessageBuilder(pnsRegistration.AppId, $"{bundle.Server}{handle}", jwt.Value)
+            var request = new ApnsMessageBuilder(pnsRegistration.AppId, $"{GetServer(pnsRegistration.Environment!.Value)}{handle}", jwt.Value)
```

Comment:
You should extract `$"{GetServer(pnsRegistration.Environment!.Value)}{handle}"` into a `buildUrl` method.